scripts/test/waitForIngest.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+# Copyright 2022 Crown Copyright
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SCRIPTS_DIR=$(cd "$(dirname "$0")" && cd .. && pwd)
+VERSION=$(cat "$SCRIPTS_DIR/templates/version.txt")
+JARS_DIR="$SCRIPTS_DIR/jars"
+GENERATED_DIR="$SCRIPTS_DIR/generated"
+
+SYSTEM_TEST_JAR="$JARS_DIR/system-test-$VERSION-utility.jar"
+WRITE_DATA_OUTPUT_FILE="$GENERATED_DIR/writeDataOutput.json"
+
+INSTANCE_PROPERTIES=${GENERATED_DIR}/instance.properties
+INSTANCE_ID=$(grep -F sleeper.id "${INSTANCE_PROPERTIES}" | cut -d'=' -f2)
+
+source "$SCRIPTS_DIR/functions/timeUtils.sh"
+START_TIME=$(record_time)
+
+echo "-------------------------------------------------------------------------------"
+echo "Waiting for tasks to generate data"
+echo "-------------------------------------------------------------------------------"
+java -cp "${SYSTEM_TEST_JAR}" \
+sleeper.systemtest.ingest.WaitForGenerateData "${WRITE_DATA_OUTPUT_FILE}"
+
+END_GENERATE_DATA_TIME=$(record_time)
+echo "Waiting for data generation finished at $(recorded_time_str "$END_GENERATE_DATA_TIME"), took $(elapsed_time_str "$START_TIME" "$END_GENERATE_DATA_TIME")"
+
+echo "-------------------------------------------------------------------------------"
+echo "Triggering ingest"
+echo "-------------------------------------------------------------------------------"
+java -cp "${SYSTEM_TEST_JAR}" \
+sleeper.systemtest.ingest.TriggerIngestFromQueue "${INSTANCE_ID}"
+
+END_TRIGGER_INGEST_TIME=$(record_time)
+echo "Triggering ingest finished at $(recorded_time_str "$END_TRIGGER_INGEST_TIME"), took $(elapsed_time_str "$END_GENERATE_DATA_TIME" "$END_TRIGGER_INGEST_TIME")"
+
+echo "-------------------------------------------------------------------------------"
+echo "Waiting for ingest tasks"
+echo "-------------------------------------------------------------------------------"
+java -cp "${SYSTEM_TEST_JAR}" \
+sleeper.systemtest.ingest.WaitForIngestTasks "${INSTANCE_ID}"
+
+FINISH_TIME=$(record_time)
```

Comment:
Since we're logging the duration of each wait step before the next wait step starts, we could remove the duplicate logging at the end of the script (so just include Started at + Overall wait time) and move the "Ingest tasks finished at" line into this wait step.

Alternatively we could make the logging the same as the other scripts, by removing the output in each wait step and having the summary of all steps at the end.

Reply:
I think for all the scripts with separate steps, where I added timing logs I logged both at the end of the step and at the end of the script. Where a step just calls a separate script, I relied on the other script to log at the end of it.

The problem is individual steps can generate a lot of logs, especially the wait scripts. You don't want to have to scroll up to find how long each step took, and you want to see the timing as soon as the step finishes.